pkg/util/result/result.go

```diff
@@ -41,10 +44,26 @@ import "github.com/pkg/errors"
 // `Error` member function can be used to turn a nullable `Result` into an
 // `error`.
 type Result struct {
-	err error
+	cause *Result
+	err   error
 }
 
-func (r *Result) Error() error { return r.err }
+// Error returns the error most appropriate for this Result. If this result represents a bailed operation, it will be
+// nil.
+func (r *Result) Error() error {
+	if r != nil {
+		return r.err
+	}
+	return nil
+}
+
+// Cause returns the Result that caused the current Result to be propegated.
```

Comment:
propegated.

add a spellchecker to vim :)

Reply:
gometalinter is my spellchecker :(